cmd/krel/cmd/promote-images.go

```diff
@@ -187,11 +182,19 @@ func runPromote(opts *promoteOptions) error {
 			return errors.Wrap(err, "while reading the current promoter image list")
 		}
 	}
+	if len(oldlist) == 0 {
+		return errors.Wrap(err, "no image list found")
+	} 
+	manifest, err := reg.ParseManifest(oldlist)
+	if err != nil {
+		return errors.Wrap(err, "while parsing manifest")
+	}
+	src := manifest.SrcRegistry
 
-	// Run cip-mm
+	// Run src-registry
```

Comment:
Hey @bytecook , thank you for looking into this!

The idea behind this issue is that you have to replace this block here to avoid calling an external binary. If you look into the command package you'll see that you're feeding data to the command line.

What you need to do is implement the manifest generation in go instead getting it from calling the binary.

Reply:
Let me take a longer look at the command package and I'll get back to you with any questions or modifications I make.

Reply:
Am I on the right track thinking of it this way:
`command.NewWithWorkDir(filepath.Join(repo.Dir(), release.GCRIOPathProd),
		fmt.Sprintf("echo %v > %s.yaml", manifest.ToRegInvImageDigest(), src))`